cli/src/commands/parse.rs

```diff
@@ -113,108 +146,49 @@ pub fn parse_depfile(
-        let packages = content.and_then(|content| parse_lockfile_content(&content, parser));
-
-        // Attempt to strip root path for identified lockfile
-        let lockfile = strip_root_path(lockfile, project_root)?;
-
-        match packages {
-            Ok(packages) => return Ok(ParsedLockfile::new(lockfile, format, packages)),
-            // Store error on failure.
-            Err(err) => lockfile_error = Some(err),
-        }
-    }
-
-    // Abort if generation is disabled or path is neither lockfile nor manifest.
-    let maybe_manifest = parser.is_path_manifest(&path);
-    if !(generate_lockfiles && maybe_manifest) {
-        // Return the original lockfile parsing error.
-        match lockfile_error {
-            // Report parsing errors only for lockfiles.
-            Some(err) if !maybe_manifest => return Err(err.into()),
-            _ => return Err(ParseError::ManifestWithoutGeneration(path)),
-        }
-    }
-
-    // If the lockfile couldn't be parsed, or there is none, we generate a new one.
-
-    // Find the generator for this format.
-    let generator = match parser.generator() {
-        Some(generator) => generator,
-        None => return Err(anyhow!("unsupported manifest file {path:?}").into()),
-    };
-
-    let display_path = strip_root_path(path.to_path_buf(), project_root)?;
-
-    eprintln!("Generating lockfile for manifest {display_path:?} using {format:?}…");
-
-    // Generate a new lockfile.
-    let generated_lockfile =
-        generate_lockfile(generator, format.name(), &path, sandbox_generation)?;
+    let display_path = strip_root_path(&path, project_root)?.display().to_string();
 
-    // Parse the generated lockfile.
-    let packages = parse_lockfile_content(&generated_lockfile, parser)?;
-
-    Ok(ParsedLockfile::new(display_path, format, packages))
-}
-
-/// Generate a lockfile from a manifest inside a sandbox.
-fn generate_lockfile(
-    generator: &dyn Generator,
-    lockfile_type: &str,
-    path: &Path,
-    sandbox: bool,
-) -> Result<String> {
-    let canonical_path = path.canonicalize()?;
-
-    if sandbox {
-        // Spawn separate sandboxed process to generate the lockfile.
-        let current_exe = env::current_exe()?;
-        let output = Command::new(current_exe)
-            .arg("generate-lockfile")
-            .arg(lockfile_type)
-            .arg(canonical_path)
-            .output()?;
+    if sandbox_generation && generate_lockfiles {
+        // Spawn separate process to allow sandboxing lockfile generation.
+        let path = path.canonicalize().map_err(anyhow::Error::from)?;
+        let lockfile_type = format.map(|format| format.to_string());
+        let mut command = parse_sandboxed_command(
+            &path,
+            &display_path,
+            lockfile_type.as_ref(),
+            generate_lockfiles,
+            false,
+        )?;
+        let output = command.output().map_err(anyhow::Error::from)?;
```

Comment:
Since stderr is now captured (and discarded on success), we've lost log messages. In particular, I think [this `eprintln!()`][1] was useful and now it is hidden.

[1]: https://github.com/phylum-dev/cli/blob/ce3a5691db1a4b0621d922c77d49834ee57b871f/lockfile/src/parse_depfile.rs#L148

Removing the log message may be good for extensions (which should control their own output). But it's a negative for the CLI itself.

### Before

```
> phylum parse Cargo.toml >/dev/null
Generating lockfile for manifest "Cargo.toml" using Cargo…
```

### After

```
> ./target/debug/phylum parse Cargo.toml >/dev/null
```